### PR TITLE
DEV-2691: Fix autofill for country field

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

This is stacked on DEV-2689 since that one revises the DDonorAddress component.

- Disables browser autofill on SearchableSelect.
- Adds autofill handling on CountrySelect.

#### Why are we doing this? How does it help us?

Lets users use autofill in their browser to complete a contribution form.

#### How should this be manually tested? Please include detailed step-by-step instructions.

This should work in Chrome and Safari. Firefox will not work, according to my own testing--when I chose "United States" as my country there, it would autofill "US" instead.

1. Visit a contribution page.
2. Use browser autofill to enter your address.
3. The country you have selected should appear in the form. The field will not change color to show it was autofilled, however.
4. Click "Continue to Payment" immediately without taking any other actions.
5. Finish a test contribution.
6. Confirm that the country you autofilled shows in the Contributions section as well as Stripe.

Also:

1. Visit a contribution page.
2. Use browser autofill to enter your address.
3. The country you have selected should appear in the form. The field will not change color to show it was autofilled, however.
4. Type in a different address manually into the Country field.
5. Confirm the field is editable as usual (shows completions, etc.)
6. Finish a test contribution.
7. Confirm that the country you entered shows in the Contributions section as well as Stripe.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

I think the whole approach here is not great :(. If we can find a way to make it work with the grain of MUI's autocomplete component, that would be great, but I wasn't able to find a way. The closest I could get is to have the value autofilled into the text field, but it would not call onChange, only onInput, and after autofill happened, the completions dropdown would appear as if the user had typed into the field but not selected an option.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-2691

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.

### CODE REVIEW

**Review by GuiMend — APPROVED** [review]
Code look good 👍 

It's not working with my autofill address that has country = Brazil (not sure this is relevant if it works with US)

File: spa/src/components/base/Select/CountrySelect.test.tsx [hunk 1]
```diff
@@ -46,6 +46,39 @@ describe('CountrySelect', () => {
...
+      expect(onChange).not.toBeCalled();
+    });
+
+    it('does nothing if onChange is not defined', () => {
+      tree();
+      expect(() => userEvent.type(screen.getByTestId('autofill-proxy'), 'BBB')).not.toThrow();
+    });
+  });
+
+  it('calls onChange if the browser autofills a matching value in the hidden input', () => {
```
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/base/Select/CountrySelect.test.tsx]: Fixed in 507724d1746d7077938476db769b19a888b394a7

**Review by GuiMend — CHANGES_REQUESTED** [review]
@nrh-cklimas, something weird is happening this time. Where did that icon come from?

image

**Review by GuiMend — APPROVED** [review]
@nrh-cklimas , we have some failling tests but the issue I saw previously isn't happening anymore 👍 

PS: The country autofill still isn't working with my brazilian address (but if it works with US we can ignore this)

**Review by x110dc — APPROVED** [review]

**Pull request merged**